pkg/controller/storagecluster/platform_detection.go

```diff
@@ -55,3 +61,18 @@ func avoidObjectStore(p configv1.PlatformType) bool {
 	}
 	return false
 }
+
+func (p *Platform) TunePlatformForFastDevices(c client.Client) (bool, error) {
```

Comment:
can you do 
```
func (r *ReconcileStorageCluster) TunePlatformForFastDevices() (bool, error) {
  platform := r.platform
  client := r.client
  ...
}
```

Makes it easier on the caller, Imho.

Reply:
done.

Reply:
I'd argue that we keep the old way.

It seems like we made it easier on the caller but we actually made it difficult.
While with the old implementation caller only required a `*Platform` and a `client.Client`, with the suggested change caller would now need to create a concrete `*ReconcileStorageCluster` even though it isn't necessary.

It is easy to ignore this as we seem to have a `*ReconcileStorageCluster` always available but we shouldn't rely on that.
Also, since this tuning is a behavior that depends on platform rather than the reconciler, it should really be a method on Platform.

Reply:
We can have an internal method on platform.
But I request the convenience wrapper on the reconciler, since it has all we need, and it avoids code duplication (like with `avoidObjectStore`) when there might be multiple callers.
This pattern of GetPlatform and then invoke the other method is cumbersome and error-prone imho.

Reply:
in the end, I won't insist too much if folks are more inline with the original approach. I think both approaches have pros and cons. ;-)